fastlane_core/lib/fastlane_core/tool_collector.rb

```diff
@@ -0,0 +1,72 @@
+module FastlaneCore
+  class ToolCollector
```

Comment:
Yes. We originally had it as part of the same set of changes, but because of the dependency between them, we needed to separate the work into two PRs. The second one will come soon.